android/modules/database/src/java/ti/modules/titanium/database/DatabaseModule.java

```diff
@@ -53,101 +53,98 @@ public DatabaseModule()
 	@Kroll.method
 	public TiDatabaseProxy open(Object file)
 	{
-		TiDatabaseProxy dbp = null;
-
-		try {
-			if (file instanceof TiFileProxy) {
-				TiFileProxy tiFile = (TiFileProxy) file;
-				String absolutePath = tiFile.getBaseFile().getNativeFile().getAbsolutePath();
-				Log.d(TAG, "Opening database from filesystem: " + absolutePath);
+		// Validate argument.
+		if (file == null) {
+			throw new IllegalArgumentException("Ti.Database.open() was given a null argument.");
+		}
 
-				SQLiteDatabase db = SQLiteDatabase.openDatabase(
-					absolutePath, null, SQLiteDatabase.CREATE_IF_NECESSARY | SQLiteDatabase.NO_LOCALIZED_COLLATORS);
+		// Attempt to create/open the given database file/name.
+		TiDatabaseProxy dbp = null;
+		if (file instanceof TiFileProxy) {
+			TiFileProxy tiFile = (TiFileProxy) file;
+			String absolutePath = tiFile.getBaseFile().getNativeFile().getAbsolutePath();
+			Log.d(TAG, "Opening database from filesystem: " + absolutePath);
+
+			SQLiteDatabase db = SQLiteDatabase.openDatabase(
+				absolutePath, null, SQLiteDatabase.CREATE_IF_NECESSARY | SQLiteDatabase.NO_LOCALIZED_COLLATORS);
+			if (db != null) {
 				dbp = new TiDatabaseProxy(db);
 			} else {
-				String name = TiConvert.toString(file);
-				SQLiteDatabase db = TiApplication.getInstance().openOrCreateDatabase(name, Context.MODE_PRIVATE, null);
+				String badPath = (absolutePath != null) ? absolutePath : "(null)";
+				badPath = "'" + badPath + "'";
+				throw new RuntimeException("SQLiteDatabase.openDatabase() returned null for path: " + badPath);
+			}
+		} else {
```

Comment:
Should we check if `file` is a `String` and return an invalid argument error if it is not?